Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,143 @@
+        /// </summary>
+        public override Type OutputType => typeof(VolumeRenkoBar);
+
+        /// <summary>
+        /// Gets the most recently consolidated piece of data. This will be null if this consolidator
+        /// has not produced any data yet.
+        /// </summary>
+        public IBaseData Consolidated
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        public event EventHandler<VolumeRenkoBar> DataConsolidated;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VolumeRenkoConsolidator"/> class using the specified <paramref name="barSize"/>.
+        /// </summary>
+        /// <param name="barSize">The constant volume size of each bar</param>
+        public VolumeRenkoConsolidator(decimal barSize)
+        {
+            _barSize = barSize;
+        }
+
+        /// <summary>
+        /// Updates this consolidator with the specified data
+        /// </summary>
+        /// <param name="data">The new data for the consolidator</param>
+        public override void Update(BaseData data)
+        {
+            var rate = data.Price;
+            var dataType = data.GetType();
+
+            decimal volume;
+            decimal open;
+            decimal high;
+            decimal low;
+            decimal close;
+
+            if (dataType == typeof(TradeBar))
+            {
+                volume = ((TradeBar)data).Volume;
+                open = ((TradeBar)data).Open;
+                high = ((TradeBar)data).High;
+                low = ((TradeBar)data).Low;
+                close = rate;
+            }
+            else if (dataType == typeof(Tick))
+            {
+                // Only include actual trade information
+                if (((Tick)data).TickType != TickType.Trade)
+                {
+                    return;
+                }
+
+                volume = ((Tick)data).Quantity;
+                open = rate;
+                high = rate;
+                low = rate;
+                close = rate;
+            }
+            else
+            {
+                throw new ArgumentException("VolumeRenkoConsolidator() must be used with TradeBar or Tick data.");
+            }
+
+            if (_firstTick)
+            {
+                _firstTick = false;
+                _currentBar = new VolumeRenkoBar(data.Symbol, data.Time, data.EndTime, _barSize, open, high, low, close, volume);
+            }
+            else
+            {
+                _volumeLeftOver = _currentBar.Update(data.EndTime, high, low, close, volume);
+                while (_volumeLeftOver >= 0)
+                {
+                    OnDataConsolidated(_currentBar);
+                    _currentBar = new VolumeRenkoBar(data.Symbol, data.EndTime, data.EndTime, _barSize, close, high, low, close, 0);
```

Comment:
> new VolumeRenkoBar(data.Symbol, data.EndTime, data.EndTime

Sending `EndTime` twice 

> close, high, low, close, 0

Sending `close` twice

- Should create the new `VolumeRenkoBar` in a single location, not 2 lines to avoid these types of issues

Reply:
I create a `Rollover` method in the `VolumeRenkoBar` class to use information from the current bar to create a new bar (like a `Clone`, but modified)